### PR TITLE
New Food Added: Cheese Wheel

### PR DESCRIPTION
**Pull request by t-toasted**
## About the PR 
Adds a large cheese wheel to the game. You can create this by putting 4 cheeses into the oven; but, beware- you cannot eat the cheese wheel before cutting it! Cutting it gives you 4 things of cheese.

Rejoice at the useless cheese wheel. Store it in your backpack for long walks in space. Throw it at people?
unknown (13)

## Why's this needed? 
It would be funny! Slightly useless, but funny. 

## Changelog

```changelog
(u)Roddy
(+)Adds a cheese wheel to the game. 
```

### CODE REVIEW

**Comment by NightmareChamillian** [comment]
silly goofy idea: can you make it so it rolls similar to bowling melons/paper airplanes?

**Comment by t-toasted** [comment]
> silly goofy idea: can you make it so it rolls similar to bowling melons/paper airplanes?

we actually looked into that and looked into spriting such a thing, but it's- well, surprisingly hard to sprite a cheesewheel in such a way. Sadly ):

**Comment by NightmareChamillian** [comment]
such a shame! ah well.

File: code/modules/food_and_drink/snacks.dm [hunk 1]
```diff
@@ -2892,3 +2892,49 @@ ABSTRACT_TYPE(/obj/item/reagent_containers/food/snacks/soup)
...
+
+/obj/item/reagent_containers/food/snacks/cheesewheel
+	name = "cheese wheel"
+	desc = "A giant wheel of cheese. It seems a slice is already missing."
+	icon = 'icons/obj/foodNdrink/food_meals.dmi'
+	icon_state = "cheesewheel"
+	inhand_image_icon = 'icons/mob/inhand/hand_food.dmi'
+	throwforce = 1
+	real_name = "cheesewheel"
+	flags = NOSPLASH | FPRINT | TABLEPASS
```
Comment (frawhst) [inline code/modules/food_and_drink/snacks.dm:2905]: I think this is redundant? These seem like default item flags so I'm not sure why you're redefining them here.

File: code/modules/food_and_drink/snacks.dm [hunk 2]
```diff
@@ -2892,3 +2892,49 @@ ABSTRACT_TYPE(/obj/item/reagent_containers/food/snacks/soup)
...
 	food_effects = list("food_hp_up","food_refreshed","food_warm")
+
+
+/obj/item/reagent_containers/food/snacks/cheesewheel
+	name = "cheese wheel"
+	desc = "A giant wheel of cheese. It seems a slice is already missing."
+	icon = 'icons/obj/foodNdrink/food_meals.dmi'
+	icon_state = "cheesewheel"
+	inhand_image_icon = 'icons/mob/inhand/hand_food.dmi'
+	throwforce = 1
```
Comment (frawhst) [inline code/modules/food_and_drink/snacks.dm:2903]: This is also default item throwforce
also it's a damn cheese wheel give it some HEFT

File: code/modules/food_and_drink/snacks.dm [hunk 3]
```diff
@@ -2892,3 +2892,49 @@ ABSTRACT_TYPE(/obj/item/reagent_containers/food/snacks/soup)
...
 	initial_reagents = "currypowder"
 	food_effects = list("food_hp_up","food_refreshed","food_warm")
+
+
+/obj/item/reagent_containers/food/snacks/cheesewheel
+	name = "cheese wheel"
+	desc = "A giant wheel of cheese. It seems a slice is already missing."
+	icon = 'icons/obj/foodNdrink/food_meals.dmi'
+	icon_state = "cheesewheel"
+	inhand_image_icon = 'icons/mob/inhand/hand_food.dmi'
```
Comment (frawhst) [inline code/modules/food_and_drink/snacks.dm:2902]: this is also default for food and doesn't need to be overridden

File: code/modules/food_and_drink/snacks.dm [hunk 4]
```diff
@@ -2892,3 +2892,49 @@ ABSTRACT_TYPE(/obj/item/reagent_containers/food/snacks/soup)
...
+				return
+
+			var/turf/T = get_turf(src)
+			user.visible_message("[user] cuts [src] into slices.", "You cut [src] into slices.")
+			var/makeslices = 4
+			while (makeslices > 0)
+				new slicetype (T)
+				makeslices -= 1
+			qdel (src)
+		else ..()
```
Comment (frawhst) [inline code/modules/food_and_drink/snacks.dm:2940]: This all shouldn't be needed, we have a system for slicing food- see 
https://github.com/goonstation/goonstation/blob/fefe1d128cdf534cbe5db9fa172a1d7e1742c41b/code/modules/chemistry/tools/food_and_drink.dm#L73

I dunno why bread isn't using that already. sucks!!

**Review by frawhst — COMMENTED** [review]
Good feature though. Can't wait to do skyrim RP and down 17 cheese wheels during combat.

**Comment by SenExus** [comment]
I LOVE CHEESE so this is perfect for me!

File: code/modules/food_and_drink/snacks.dm (hunk 1 above)
Comment (t-toasted) [reply code/modules/food_and_drink/snacks.dm:2905]: > I think this is redundant? These seem like default item flags so I'm not sure why you're redefining them here.

I also wasn't sure what they did! I just saw that others had them. Removed!

File: code/modules/food_and_drink/snacks.dm (hunk 3 above)
Comment (t-toasted) [reply code/modules/food_and_drink/snacks.dm:2902]: > this is also default for food and doesn't need to be overridden

Noted! It is gone

**Review by flrsh — CHANGES_REQUESTED** [review]

File: code/datums/cookingrecipes.dm [hunk 5]
```diff
@@ -1825,6 +1825,15 @@ ABSTRACT_TYPE(/datum/cookingrecipe)
 	cookbonus = 14
 	output = /obj/item/reagent_containers/food/snacks/curry_udon_bowl
 
+
+/datum/cookingrecipe/cheesewheel
+	item1= /obj/item/reagent_containers/food/snacks/ingredient/cheese
```
Comment (ZeWaka) [inline code/datums/cookingrecipes.dm:1830]:
```suggestion
	item1 = /obj/item/reagent_containers/food/snacks/ingredient/cheese
```

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**